src/algorithms/bbvi.jl

```diff
@@ -0,0 +1,59 @@
+"""
+$(TYPEDEF)
+
+Black-Box Variational Inference (BBVI) with automatic differentiation
+backend `AD`.
+
+# Fields
+
+$(TYPEDFIELDS)
+"""
+struct BBVI{AD} <: VariationalInference{AD}
```

Comment:
It's awesome to see BBVI here. I didn't see any implementations for rao-blackwellization / control-variates to reduce variance of the noisy gradient. But I suppose this could be extended.

Reply:
That's indeed my intention, I just thought it would be good to have at least two VI implementations! Control variates and rao Blackwell will follow :)